Request for non-existing /index endpoints with size parameter yields 500 status

```
❯ curl -v -X 'GET' 'https://service.azul.data.humancellatlas.org/index/donors?size=1'
Note: Unnecessary use of -X or --request, GET is already inferred.
* Host service.azul.data.humancellatlas.org:443 was resolved.
* IPv6: (none)
* IPv4: 99.84.41.77, 99.84.41.46, 99.84.41.58, 99.84.41.60
*   Trying 99.84.41.77:443...
* Connected to service.azul.data.humancellatlas.org (99.84.41.77) port 443
* ALPN: curl offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
* (304) (IN), TLS handshake, Unknown (8):
* (304) (IN), TLS handshake, Certificate (11):
* (304) (IN), TLS handshake, CERT verify (15):
* (304) (IN), TLS handshake, Finished (20):
* (304) (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / AEAD-AES128-GCM-SHA256 / [blank] / UNDEF
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=service.azul.data.humancellatlas.org
*  start date: Jun 14 00:00:00 2025 GMT
*  expire date: Jul 11 23:59:59 2026 GMT
*  subjectAltName: host "service.azul.data.humancellatlas.org" matched cert's "service.azul.data.humancellatlas.org"
*  issuer: C=US; O=Amazon; CN=Amazon RSA 2048 M02
*  SSL certificate verify ok.
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://service.azul.data.humancellatlas.org/index/donors?size=1
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: service.azul.data.humancellatlas.org]
* [HTTP/2] [1] [:path: /index/donors?size=1]
* [HTTP/2] [1] [user-agent: curl/8.7.1]
* [HTTP/2] [1] [accept: */*]
> GET /index/donors?size=1 HTTP/2
> Host: service.azul.data.humancellatlas.org
> User-Agent: curl/8.7.1
> Accept: */*
>
* Request completely sent off
< HTTP/2 500
< content-type: application/json
< content-length: 128
< date: Tue, 23 Sep 2025 17:59:42 GMT
< x-amzn-trace-id: Root=1-68d2e00e-4b35edd3498936fc12a188bf;Parent=7ccff6684c35093b;Sampled=0;Lineage=1:2808a99a:0
< x-amzn-requestid: 7bd4f220-f96b-48dc-b9cf-32a0e18b268b
< referrer-policy: strict-origin-when-cross-origin
< x-xss-protection: 1; mode=block
< access-control-allow-origin: *
< strict-transport-security: max-age=63072000; includeSubDomains; preload
< access-control-allow-headers: Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key
< x-frame-options: DENY
< content-security-policy: default-src 'self';img-src 'self' data:;script-src 'self';style-src 'self';frame-ancestors 'none';form-action 'self'
< x-amz-apigw-id: RXfyQHC4IAMEIUg=
< cache-control: no-store
< x-content-type-options: nosniff
< x-cache: Error from cloudfront
< via: 1.1 d2bbb81f4893c1006c27c11f151a19ca.cloudfront.net (CloudFront)
< x-amz-cf-pop: LAX54-P11
< x-amz-cf-id: Jg2eRF6wunJypijjxjLKW82uA--R8ZUkyof4TXrYZEPgkJcGtCWgNA==
<
* Connection #0 to host service.azul.data.humancellatlas.org left intact
{"Code":"InternalServerError","Message":"An internal server error occurred.","RequestId":"bd0c99aa-9c2e-46de-ad00-0d2e7e31836d"}
```

**CloudWatch Logs Insights**    
region: us-east-1    
log-group-names: /aws/lambda/azul-service-prod    
start-time: -3600s    
end-time: 0s    
query-string:
  ```
  fields @timestamp, @message
| filter @requestId like /bd0c99aa-9c2e-46de-ad00-0d2e7e31836d/
| sort @timestamp asc
| limit 200
  ```
---
| @timestamp | @message |
| --- | --- |
| 2025-09-23 17:59:42.101 | START RequestId: bd0c99aa-9c2e-46de-ad00-0d2e7e31836d Version: $LATEST |
| 2025-09-23 17:59:42.101 | [INFO] 2025-09-23T17:59:42.101Z bd0c99aa-9c2e-46de-ad00-0d2e7e31836d azul.chalice Received GET request for &#x27;&#x2f;index&#x2f;donors&#x27;, with {&quot;query&quot;: {&quot;size&quot;: &quot;1&quot;}, &quot;headers&quot;: {&quot;accept&quot;: &quot;*&#x2f;*&quot;, &quot;cloudfront-forwarded-proto&quot;: &quot;https&quot;, &quot;cloudfront-is-desktop-viewer&quot;: &quot;true&quot;, &quot;cloudfront-is-mobile-viewer&quot;: &quot;false&quot;, &quot;cloudfront-is-smarttv-viewer&quot;: &quot;false&quot;, &quot;cloudfront-is-tablet-viewer&quot;: &quot;false&quot;, &quot;cloudfront-viewer-asn&quot;: &quot;20115&quot;, &quot;cloudfront-viewer-country&quot;: &quot;US&quot;, &quot;host&quot;: &quot;service.azul.data.humancellatlas.org&quot;, &quot;user-agent&quot;: &quot;curl&#x2f;8.7.1&quot;, &quot;via&quot;: &quot;2.0 d2bbb81f4893c1006c27c11f151a19ca.cloudfront.net (CloudFront)&quot;, &quot;x-amz-cf-id&quot;: &quot;Jg2eRF6wunJypijjxjLKW82uA--R8ZUkyof4TXrYZEPgkJcGtCWgNA==&quot;, &quot;x-amzn-trace-id&quot;: &quot;Root=1-68d2e00e-4b35edd3498936fc12a188bf&quot;, &quot;x-forwarded-for&quot;: &quot;71.84.86.94, 3.172.116.106&quot;, &quot;x-forwarded-port&quot;: &quot;443&quot;, &quot;x-forwarded-proto&quot;: &quot;https&quot;}}. |
| 2025-09-23 17:59:42.101 | [INFO] 2025-09-23T17:59:42.101Z bd0c99aa-9c2e-46de-ad00-0d2e7e31836d azul.chalice … without a request body |
| 2025-09-23 17:59:42.101 | [INFO] 2025-09-23T17:59:42.101Z bd0c99aa-9c2e-46de-ad00-0d2e7e31836d azul.chalice Did not authenticate request. |
| 2025-09-23 17:59:42.103 | [ERROR] 2025-09-23T17:59:42.102Z bd0c99aa-9c2e-46de-ad00-0d2e7e31836d azul-service-prod Caught exception for path &#x2f;index&#x2f;{entity_type} Traceback (most recent call last):   File &quot;&#x2f;var&#x2f;task&#x2f;chalice&#x2f;app.py&quot;, line 1922, in _get_view_function_response     response = view_function(**function_args)                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^   File &quot;&#x2f;var&#x2f;task&#x2f;app.py&quot;, line 1002, in repository_search     validate_repository_search(entity_type, query_params)   File &quot;&#x2f;var&#x2f;task&#x2f;app.py&quot;, line 494, in validate_repository_search     validate_params(params, **{   File &quot;&#x2f;var&#x2f;task&#x2f;azul&#x2f;service&#x2f;app_controller.py&quot;, line 151, in validate_params     validator(param_value)   File &quot;&#x2f;var&#x2f;task&#x2f;app.py&quot;, line 519, in validate_size     sorting = app.metadata_plugin.exposed_indices[entity_type]               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^ KeyError: &#x27;donors&#x27; |
| 2025-09-23 17:59:42.103 | [INFO] 2025-09-23T17:59:42.103Z bd0c99aa-9c2e-46de-ad00-0d2e7e31836d azul.chalice Returning 500 response with headers {&quot;headers&quot;: {&quot;Access-Control-Allow-Origin&quot;: &quot;*&quot;, &quot;Access-Control-Allow-Headers&quot;: &quot;Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key&quot;, &quot;Content-Security-Policy&quot;: &quot;default-src &#x27;self&#x27;;img-src &#x27;self&#x27; data:;script-src &#x27;self&#x27;;style-src &#x27;self&#x27;;frame-ancestors &#x27;none&#x27;;form-action &#x27;self&#x27;&quot;, &quot;Referrer-Policy&quot;: &quot;strict-origin-when-cross-origin&quot;, &quot;Strict-Transport-Security&quot;: &quot;max-age=63072000; includeSubDomains; preload&quot;, &quot;X-Content-Type-Options&quot;: &quot;nosniff&quot;, &quot;X-Frame-Options&quot;: &quot;DENY&quot;, &quot;X-XSS-Protection&quot;: &quot;1; mode=block&quot;, &quot;Cache-Control&quot;: &quot;no-store&quot;}}. |
| 2025-09-23 17:59:42.103 | [INFO] 2025-09-23T17:59:42.103Z bd0c99aa-9c2e-46de-ad00-0d2e7e31836d azul.chalice … with a response body of length 133 being {&quot;Code&quot;: &quot;InternalServerError&quot;, &quot;Message&quot;: &quot;An internal server error occurred.&quot;, &quot;RequestId&quot;: &quot;bd0c99aa-9c2e-46de-ad00-0d2e7e31836d&quot;} |
| 2025-09-23 17:59:42.105 | END RequestId: bd0c99aa-9c2e-46de-ad00-0d2e7e31836d |
| 2025-09-23 17:59:42.105 | REPORT RequestId: bd0c99aa-9c2e-46de-ad00-0d2e7e31836d Duration: 3.65 ms Billed Duration: 4 ms Memory Size: 2048 MB Max Memory Used: 182 MB |
---

```python
Traceback (most recent call last):
  File "/var/task/chalice/app.py", line 1922, in _get_view_function_response
    response = view_function(**function_args)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/app.py", line 1002, in repository_search
    validate_repository_search(entity_type, query_params)
  File "/var/task/app.py", line 494, in validate_repository_search
    validate_params(params, **{
  File "/var/task/azul/service/app_controller.py", line 151, in validate_params
    validator(param_value)
  File "/var/task/app.py", line 519, in validate_size
    sorting = app.metadata_plugin.exposed_indices[entity_type]
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
KeyError: 'donors'
```

--- a/src/azul/service/index_controller.py
+++ b/src/azul/service/index_controller.py
@@ -347,6 +347,13 @@ class IndexController(QueryController):
     def search(self, entity_type: str, entity_id: str | None = None) -> str | JSON:
         request = self.current_request
         query_params = self._hoist_parameters(request)
+        # Validate `catalog` first since `_validate_entity_type` requires a
+        # valid catalog to load the metadata plugin.
+        validate_params(query_params,
+                        catalog=self._validate_catalog,
+                        allow_extra_params=True)
+        # Likewise, `_validate_size` requires a valid entity type.
+        self._validate_entity_type(entity_type)
         validate_params(query_params,
                         catalog=self._validate_catalog,
                         filters=self._validate_filters,
@@ -357,7 +364,6 @@ class IndexController(QueryController):
                         search_before_uid=str,
                         size=partial(self._validate_size, entity_type),
                         sort=self._validate_field)
-        self._validate_entity_type(entity_type)
         filters = query_params.get('filters')
         pagination = self._pagination(entity_type)
         authentication = self._authentication(request)

